src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,130 @@
+    int totalProducts = 0;
+    int totalReviews = 0;
+    int totalUsers = 0;
+    int productAmount;
+    int userAmount;
+    HashMap<String, Integer> users;
+    HashMap<String, Integer> products;
+    HashMap<Integer, String> revertProducts;
+    //Create a new csv file to push data with specified path
+
+    MovieRecommender(String pathFile) throws IOException {
+        users = new HashMap<String, Integer>();
+        products = new HashMap<String, Integer>();
+        revertProducts = new HashMap<Integer, String>();
+        readFile(pathFile);
+    }
+
+    private void readFile(String pathFile) throws IOException {
+
+        String productID = "product/productId: ";
+        String userID = "review/userId: ";
+        String valueID = "review/score: ";
+        String fileName = pathFile;
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(
+                new FileInputStream(fileName), StandardCharsets.UTF_8))) {
+            String line;
+            String output = "data/output.csv";
+            Path path = Paths.get(output);
+            try (BufferedWriter writer = Files.newBufferedWriter(path, StandardCharsets.UTF_8)) {
+
+                while ((line = br.readLine()) != null) {  //Checking if the line contains a specific string to match
+                    if (line.startsWith(productID)) {
+                        currentProduct = line.replace(productID, ""); //B003AI2VGA
+                        totalMovies++;
+                        if (!products.containsKey(currentProduct)) {
+                            totalProducts++;
+                            productAmount = totalProducts;
+                            products.put(currentProduct, totalProducts);
+                            revertProducts.put(totalProducts, currentProduct);
+                        } else {
+                            productAmount = products.get(currentProduct);
+                        }
+                    } else if (line.startsWith(userID)) {
+                        currentUser = line.replace(userID, "");
+                        if (!users.containsKey(currentUser)) {
+                            totalUsers++;
+                            userAmount = totalUsers;
+                            users.put(currentUser, totalUsers);
+//                        System.out.println(users.size());
+                        } else {
+                            userAmount = users.get(currentUser);
+                        }
+                    } else if (line.startsWith(valueID)) {
+                        currentValue = line.replace(valueID, ""); // 5.0
+//                  System.out.println(currentProduct + currentUser + currentValue);
+                        writer.write(userAmount + "," + productAmount + "," + currentValue + "\n");
+                        totalReviews++;
+                    }
+                }
+                br.close();
+                writer.close();
+            }catch (IOException e) {
```

Comment:
not sure if we need to wrap all under two try catch for the same exception that is already being catch on the method signature 🤔